### PR TITLE
[doc] adds the missing documentation of the server.method settings

### PR DESCRIPTION
**Pull request by return42**
TL;DR; For all the issues that comes with HTTP POST I recommend instance maintainers to switch to GET and lock the property in the preferences:

```yaml
server:
  method: GET

preferences:
  lock:
    - method
```

We don't want this in the defaults of the SearXNG distributions for the pro vs cons reasons named in the discussion below: 

-----------

HTTP GET vs POST
================

Already diskussed in "Method POST harms UX without providing a tangible privacy benefit" [1].  At that time we agreed to POST.  However, as we are having more and more drawbacks with POST, I suggest that we reconsider our previous decision. The latest cause was [2], but we also have other problems that negatively affect the UI [1]:

* Open Link in New Tab: does not work for our search result tabs because they aren't actual links.

* Bookmarking: a search becomes more difficult with POST. You cannot just press a browser keyboard shortcut `Ctrl+D` ... no you have to copy the Search URL from the sidebar. This can pose a real struggle for less technically-minded users.

* Sharing: a search with somebody else becomes more difficult with POST (for the same reason).

* [While autocompletion in the search field of SearXNG sends a cookie, the autocompletion in the URL bar does not send a cookie.](https://github.com/searxng/searxng/issues/1889#issuecomment-1565366772)
* https://github.com/searxng/searxng/pull/2333#issuecomment-1565392120
* [POST vs GET --> some broswers don't like HTTP POST requests in a OpenSearch query ](https://github.com/searxng/searxng/issues/431)

* https://github.com/searxng/searxng/issues/3590#issuecomment-2196891552

Lock HTTP method in the preferences
===================================

If the user changes the HTTP method in his settings, e.g. from GET to POST, but has not removed the SearXNG instance from the WEB browser and added it again, the WEB browser will continue to work with the old setting (GET), while entries in the HTML form use the newly set method (POST). Not realted to this commit, but this complication is also known from autocomplete[3].

Only very few maintainers are aware of this fact and probably none of the users know about it.  We should provide a setup in our defaults that is manageable in its entirety and comprehensible for the user.  For this reason, the option to select the HTTP method in the preferences is also disabled in this commit.

- [1] https://github.com/searxng/searxng/issues/711
- [2] https://github.com/searxng/searxng/issues/3590
- [3] https://github.com/searxng/searxng/pull/2333#issuecomment-1565392120

### CODE REVIEW

**Comment by unixfox** [comment]
No, I'm against it. Already explained why here: https://github.com/searxng/searxng/issues/711#issuecomment-1008331605

> Sharing: a search with somebody else becomes more difficult with POST (for the same reason).

`Search URL` section already exist for that. But since you hide it a while ago, people may not find it...

> Bookmarking: a search becomes more difficult with POST. You cannot just press a browser keyboard shortcut Ctrl+D ... no you have to copy the Search URL from the sidebar. This can pose a real struggle for less technically-minded users.

We could have a button to easily add the page as a bookmark: https://buffernow.com/create-bookmark-link-using-javascript/

> Open Link in New Tab: does not work for our search result tabs because they aren't actual links.

What's that? Can you elaborate a bit more?

**Comment by Bnyro** [comment]
@unixfox see https://github.com/searxng/searxng/issues/3590#issue-2364139446

**Comment by unixfox** [comment]
> @unixfox see [#3590 (comment)](https://github.com/searxng/searxng/issues/3590#issue-2364139446)

Ok. It's a very niche feature. That's not a valid reason for migrating EVERYONE to get method.

**Comment by return42** [comment]
> No, I'm against it. Already explained why here: [#711 (comment)](https://github.com/searxng/searxng/issues/711#issuecomment-1008331605)

I think the arguments @unixfox give in the linked comment are valid.  On the other hand, I think the UX arguments are also valid.  I'm torn; 

I think we should leave this PR open for now and not merge it.  The open PR can invite further readers to discuss ... or serve maintainers as a template for a setup of their instance / I switched my instance to GET years ago for the reasons mentioned above and looked the HTTP method in the preferences.  Issue https://github.com/searxng/searxng/issues/3590 has also confirmed that the settings are ideal .. at least for me :-)

**Comment by ACR-Jeff** [comment]
Why not leave it the way it is and more secure for the users, Isn't that what Searxng is all about being secure? I know why some may want GET over POST, But that brings a security issue, I have defaults set for that very reason, To keep it secure for the users, I also have one system that I have GET set, Due to needing the bookmarks, Etc, That can easily be done in the user preferences, Still giving them security and the choice otherwise.

**Comment by dalf** [comment]
Since the project became more popular, the struggle has always been the same: privacy or UX.

The moto says "a privacy-respecting, open metasearch engine", so I would stick to the POST method for the reasons explained by @unixfox 

An idea: a ~wizard~ two buttons to choose between privacy and UX in the preferences (in the first visible tab):
* Enable privacy preferences
* Enable user friendly preferences

With a tooltip, a label or something else to explain the role of each buttons.

They can be displayed and implemented in JS: user without JS won't bother about these buttons especially if the defaults are set to preserve privacy.

An alternative is to implement a real wizard with different steps and details description of each choice. Same it can be implemented in JS. A button on the top right corner can be displayed in the front page when the user don't have any preferences (no cookies).

Quick mock-up:
image

**Comment by ACR-Jeff** [comment]
Privacy is what Searxng has always been about, Yet they still have the option in preferences to change to the alternative, I use it when I need it. But IMO the alternative shouldn't be the default. Stay with what searxng has always stood for.

**Comment by Aetherinox** [comment]
I like @dalf's proposal. POST is honoring the promise of ensuring privacy. Yes, it has drawbacks, but simply introducing a way to educate the user between the two options allows them to choose if they want the privacy benefits, or if they're willing to give up a bit of privacy over a bit more of a user-friendly experience.

Personally, I stick to POST.

**Comment by omentic** [comment]
I strongly prefer GET. Aside from being how conventional search engines operate, I am not convinced that there are any tangible privacy drawbacks: private tabs + browser history are a widely understood concept, and any implications that it does not log to servers are misleading since an instance is under full control of the instance owner (and adding POST logging is trivial).

But I also don't have a stake in the matter! It is well-advertised as a configurable default, and the descriptions of each option's behavior is clear enough, IMO. People checking out the project will wind up on instances not run by the SearXNG team so I don't think that changing the default matters all too much.

**Comment by return42** [comment]
Related: 

- https://github.com/searxng/searxng/discussions/4262#discussioncomment-12005746

**Comment by return42** [comment]
> I like @dalf's proposal. POST is honoring the promise of ensuring privacy. Yes, it has drawbacks, but simply introducing a way to educate the user between the two options allows them to choose if they want the privacy benefits, or if they're willing to give up a bit of privacy over a bit more of a user-friendly experience.

I have now revised the PR, the defaults will not be changed ... but documentation will be added (f4250b6d40) .. is that OK for you all? ... can we merge the PR? .. we need a documentation of the `server.method` settings in any case :-)

**Comment by unixfox** [comment]
@return42 Can you change the PR title accordingly then?

**Comment by return42** [comment]
> @return42 Can you change the PR title accordingly then?

Thanks for reminder / done :+1:
